notebooks/tutorials/overland_flow/overland_flow_erosion/ol_flow_erosion_components.ipynb

```diff
@@ -0,0 +1,489 @@
+   "metadata": {},
+   "source": [
+    "And also instantiate a `DepthSlopeProductErosion` component:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "grid.add_field('slope_magnitude', slp_mag, at='node', clobber=True)\n",
+    "dspe = DepthSlopeProductErosion(grid,\n",
+    "                                k_e=k_e,\n",
+    "                                tau_crit=tau_c,\n",
+    "                                slope='slope_magnitude'\n",
+    "                               )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we'll make a copy of the starting terrain for later comparison, then run overland flow and erosion:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "starting_elev = elev.copy()\n",
+    "\n",
+    "for i in range(num_steps):\n",
+    "    olflow.run_one_step(dt)\n",
+    "    dspe.run_one_step(dt)\n",
+    "    slp_mag[:], slp_x, slp_y = slope_magnitude_at_node(grid, elev)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can visualize the instantaneous erosion rate at the end of the run, in m/s:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "imshow_grid(grid, dspe._E, colorbar_label='erosion rate (m/s)')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can also inspect the cumulative erosion during the event by differencing the before and after terrain:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "imshow_grid(grid, starting_elev - elev, colorbar_label='cumulative erosion (m)')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that because this is a bumpy DEM, much of the erosion has occurred on (probably digital) steps in the channels. But we can see some erosion across the slopes as well."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## `DetachmentLtdErosion`\n",
+    "\n",
+    "This component is similar to `DepthSlopeProductErosion` except that it calculates erosion rate from discharge and slope rather than depth and slope. The vertical incision rate, $I$ (equivalent to $E$ in the above; here we are following the notation in the component's documentation) is:\n",
+    "\n",
+    "$$I = K Q^m S^n - I_c$$\n",
+    "\n",
+    "where $K$ is an erodibility coefficient (with dimensions of velocity per discharge$^m$; specified by parameter `K_sp`), $Q$ is volumetric discharge, $I_c$ is a threshold with dimensions of velocity, and $m$ and $n$ are exponents that configure the physical meaning of the first term on the right side.\n",
```

Comment:
"configure the physical meaning of the first term on the right side" is somewhat confusing wording. Maybe we could just mention that they relate to things like climate, drainage basin geometry, and so forth? Not sure how much explanation this needs.

Reply:
changed this wording